charms/kserve-controller/src/charm.py

```diff
@@ -166,9 +165,6 @@ def __init__(self, *args):
         # Generate self-signed certificates and store them
         self._gen_certs_if_missing()
 
-        self._rbac_proxy_container_name = "kube-rbac-proxy"
-        self.rbac_proxy_container = self.unit.get_container(self._rbac_proxy_container_name)
```

Comment:
You removed the solution that addressed the issue of the solution itself being unaffected when the image goes away. However, it's important to note that the image had a specific purpose: protecting the metrics endpoint.

Now that you've removed the image, how are you ensuring the protection of the metrics endpoint?
ONLY remove the image does not really address the requirements IHMO

Reply:
I described the problem in the issue ... the thing is the image was never used so there is no point to keep it